### DISCUSSION TITLE
Typed Route Path Parameters Nuxt

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [posva/unplugin-vue-router#248] [Questions and Help] [answered]
Hi,

here to type the path parameter is `(to as RouteLocationNormalizedLoaded<'/path/[param]'>).params.param)`
while in Nuxt seems to be `(to as RouteLocationNormalizedLoaded<'path-param'>).params.param)` (because it's what it auto-completes).

Is this by design, or there are still work to do?

I have `~/pages/app/[name].vue`, then in a Nuxt middleware:

`(to as RouteLocationNormalizedLoaded<'app-name'>).params.name)`

because first tried with:

`(to as RouteLocationNormalizedLoaded<'/app/[name]'>).params.name)`

and didn't work.

(I'm aware typed routes is experimental in Nuxt).

Thanks.

**Comment by posva** [comment, marked as answer]
It’s because in Nuxt, routes were already given a name and this format was kept instead of using this plugin’s format. In the future the will probably be an option to use this plugin’s name format
